channels/ON-CHAIN.md

```diff
@@ -363,15 +375,144 @@ channel.
 - `state_hash`: last published state_hash 
 - `round`: last published round
-- `closes_at`: on-chain channel closing height
 - `lock_period`: agreed upon locking period by peers
+- `closes_at`: on-chain channel closing height
+- `force_blocked_until`: on-chain channel height after which a new force
+  progress can be included in a block
 
-Keeping track of the `state_hash`, `round`, `closes_at` and `lock_period` is
+Keeping track of the `state_hash`, `round`, `closes_at`, `force_blocked_until` and `lock_period` is
 necessary for nodes to be able to assess the validity of `channel_slash` and
 `channel_settle` transactions.
 Serialization defined [here](../serializations.md#channel)
+
+## Force progress on-chain
+
+Forcing progress is the mechanism to be used when a dispute
+arises between parties and one of them wants to use the blockchain as an
+arbiter. The poster provides enough state so that an off-chain contract can be
+executed on-chain and thus producing the channel's next off-chain state.
+
+This can happen both while a channel is closing or while it is still active. If the
+channel is not closing - participants can continue using it from the
+on-chain produced state or initiate a closing. If the channel is already
+closing, the force-progress updates what are the currently expected closing
+amounts for each participant (according to the contract's execution).
+
+The force progress is based on what is considered to be the latest off-chain
+state. We have no way of proving that this is the case so there is a time frame
+provided for the other participant to dispute it. This time frame is
+measured in block height and it works exactly like the `lock_period` does for
+closing sequences - a new force progress can not be posted until a certain chain
+height is reached. It is worth mentioning that what is to be
+disputed is the off-chain state that the force progress had been based on but not
+the forcing of progress itself. If an older state had been provided by the
+forcing party, the other party can post a newer co-signed off-chain state (via a
+snapshot for example). The co-signed state with the same or greater round will
+replace the on-chain produced one. This is to address the possible attack of
+a malicious actor making a couple of subsequent force pushes so even if one
+started from some old state, eventually a `round` number is produced greater
+than the one co-signed off-chain.
+
+### `channel_force_progress_tx`
+
+The `channel_force_progress_tx` can be included in a block if:
+
+- the last on-chain transaction for the targeted channel is NOT a force progress one
+- the last on-chain transaction for the targeted channel had been a force progress but the block height
+  timer had passed
+
+It consists of:
+
+- `channel_id`: channel id as recorded on-chain
+- `from`: participant of the channel that posts the force progress transaction
+- `payload`: empty or a transaction proving that the proof of inclusion is part of the channel
```

Comment:
@knarz `proof of inclusion` is a structure that _proves_ that the contract is present in a MPT with the same root hash. The `payload` is co-signed and it provides both a `channel_id` and a `state_hash`. If those hashes (PoI and `state_hash`) match - we can safely asume that this contract and its state had been part of a channel's state at least at some point of time. This might had also been some off-chain updates old and not the latest off-chain state.
The `payload` can also be empty. In this case we are using the latest on-chain persisted `state_hash` for this channel.

I'd be happy to rename the `payload` to something better fitting.